mmv1/third_party/terraform/framework_utils/framework_config.go.erb

```diff
@@ -0,0 +1,744 @@
+	creds := GetCredentials(ctx, data, initialCredentialsOnly, diags)
+
+	return creds.TokenSource
+}
+
+// GetCredentials gets credentials with a given scope (clientScopes).
+// If initialCredentialsOnly is true, don't follow the impersonation
+// settings and return the initial set of creds instead.
+func GetCredentials(ctx context.Context, data ProviderModel, initialCredentialsOnly bool, diags *diag.Diagnostics) googleoauth.Credentials {
+	var clientScopes []string
+	var delegates []string
+
+	d := data.Scopes.ElementsAs(ctx, &clientScopes, false)
+	diags.Append(d...)
+	if diags.HasError() {
+		return googleoauth.Credentials{}
+	}
+
+	d = data.ImpersonateServiceAccountDelegates.ElementsAs(ctx, &delegates, false)
+	diags.Append(d...)
+	if diags.HasError() {
+		return googleoauth.Credentials{}
+	}
+
+	if !data.AccessToken.IsNull() {
+		contents, _, err := pathOrContents(data.AccessToken.ValueString())
+		if err != nil {
+			diags.AddError("error loading access token", err.Error())
+			return googleoauth.Credentials{}
+		}
+
+		token := &oauth2.Token{AccessToken: contents}
+		if !data.ImpersonateServiceAccount.IsNull() && !initialCredentialsOnly {
+			opts := []option.ClientOption{option.WithTokenSource(oauth2.StaticTokenSource(token)), option.ImpersonateCredentials(data.ImpersonateServiceAccount.ValueString(), delegates...), option.WithScopes(clientScopes...)}
+			creds, err := transport.Creds(context.TODO(), opts...)
+			if err != nil {
+				diags.AddError("error impersonating credentials", err.Error())
+				return googleoauth.Credentials{}
+			}
+			return *creds
+		}
+
+		log.Printf("[INFO] Authenticating using configured Google JSON 'access_token'...")
+		log.Printf("[INFO]   -- Scopes: %s", clientScopes)
+		return googleoauth.Credentials{
+			TokenSource: staticTokenSource{oauth2.StaticTokenSource(token)},
+		}
+	}
+
+	if !data.Credentials.IsNull() {
+		contents, _, err := pathOrContents(data.Credentials.ValueString())
+		if err != nil {
+			diags.AddError(fmt.Sprintf("error loading credentials: %s", err), err.Error())
+			return googleoauth.Credentials{}
+		}
+
+		if !data.ImpersonateServiceAccount.IsNull() && !initialCredentialsOnly {
+			opts := []option.ClientOption{option.WithCredentialsJSON([]byte(contents)), option.ImpersonateCredentials(data.ImpersonateServiceAccount.ValueString(), delegates...), option.WithScopes(clientScopes...)}
+			creds, err := transport.Creds(context.TODO(), opts...)
+			if err != nil {
+				diags.AddError("error impersonating credentials", err.Error())
+				return googleoauth.Credentials{}
+			}
+			return *creds
+		}
+
+		creds, err := googleoauth.CredentialsFromJSON(ctx, []byte(contents), clientScopes...)
+		if err != nil {
+			diags.AddError("unable to parse credentials", err.Error())
+			return googleoauth.Credentials{}
+		}
+
+		log.Printf("[INFO] Authenticating using configured Google JSON 'credentials'...")
```

Comment:
Believe you could use [tflog](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-log/tflog#Info) in place of `log.Printf`.

For instance:
```
tflog.Info(ctx, "Authenticating using configured Google JSON 'credentials'...")
```